test/Infrastructure/LeanCode.Serialization.Tests/JsonStringEnumConvertersTests.cs

```diff
@@ -96,10 +96,120 @@ public void Snake_case_upper_converter_correctly_converts_enum_name(bool useSour
         Assert.Equal(@"""EXAMPLE_VALUE""", result);
     }
 
+    [JsonConverter(typeof(JsonCamelCaseStringEnumConverter<CamelCaseGeneric>))]
+    public enum CamelCaseGeneric
+    {
+        ExampleValue
+    }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void Camel_case_generic_converter_correctly_converts_enum_name(bool useSourceGenerationContext)
+    {
+        var result = useSourceGenerationContext
+            ? JsonSerializer.Serialize(
+                CamelCaseGeneric.ExampleValue,
+                StringEnumConvertersContext.Default.CamelCaseGeneric
+            )
+            : JsonSerializer.Serialize(CamelCaseGeneric.ExampleValue);
+
+        Assert.Equal(@"""exampleValue""", result);
+    }
+
+    [JsonConverter(typeof(JsonKebabCaseLowerStringEnumConverter<KebabCaseLowerGeneric>))]
+    public enum KebabCaseLowerGeneric
+    {
+        ExampleValue
+    }
+
+    [Theory]
+    [InlineData(true)]
```

Comment:
This is, IMO, the wrong usage for theories - these should be two separate tests as these test two separate codepaths.

Reply:
This will become (imo, unnecessarily) verbose but ye, you're right :disappointed: